custom_components/tuya_local/devices/smart_weather_station.yaml

```diff
@@ -0,0 +1,93 @@
+name: Smart Weather Station
+products:
+  - id: 000001bbfd
+    name: Smart Weather Station
+primary_entity:
+  entity: sensor
+  name: local temperature
```

Comment:
HA naming convention: capitalise the first word of an entity name.
I'm not sure whether "local" is the right word, it implies maybe a web service used to get the local temperature for your area. Many other devices use "Internal" for built in sensors where they also support external probes, which is probably clearer.

Reply:
This is how sensor is named in app. I also do not get naming convention. I mean I see plendy of devices having without first letter capitalized.

Reply:
The convention is relatively recent, you may be looking at older integrations that are not yet updated.

https://developers.home-assistant.io/blog/2022/07/10/entity_naming/